### PR TITLE
don't reduce integer literal tokens that have digit seperators

### PR DESCRIPTION
**Pull request by TyOverby**

### CODE REVIEW

**Comment by mattwar** [comment]
@dotnet-bot test windows_debug_unit32_prtest please

**Comment by TyOverby** [comment]
@dotnet/roslyn-compiler @dotnet/roslyn-ide 

Review at your leasure

**Comment by cston** [comment]
LGTM

**Comment by CyrusNajmabadi** [comment]
👍

File: src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/ReduceTokensCodeCleanupProvider.vb [hunk 1]
```diff
@@ -88,7 +88,7 @@ Namespace Microsoft.CodeAnalysis.CodeCleanup.Providers
 
                         Dim base = literal.GetBase()
 
-                        If Not base.HasValue Then
+                        If Not base.HasValue OrElse idText.Contains("_") Then
```
Comment (brettfo) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/ReduceTokensCodeCleanupProvider.vb]: Nit: Is there a character constant already defined somewhere that can be used?

Comment (brettfo) [inline src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/ReduceTokensCodeCleanupProvider.vb]: Add a comment explaining what's going on here.  If someone unfamiliar with VB or digit separators reads this code, they might not know what's happening.

File: src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests.cs [hunk 2]
```diff
@@ -2024,6 +2024,22 @@ End Module
...
 
+        [Fact]
+        [WorkItem(14034, "https://github.com/dotnet/roslyn/issues/14034")]
+        [Trait(Traits.Feature, Traits.Features.ReduceTokens)]
+        public async Task ReduceIntegersWithDigitSeparators()
+        {
+            var source = @"
+Module Module1
+    Sub Main()
+        Dim x = 100_000
```
Comment (jasonmalinowski) [inline src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests.cs]: This should have been just `0_0` to make it look like the emoji or whatever it's called these days.

Comment (TyOverby) [inline src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests.cs]: 😎

**Pull request merged**